platform/android/export/export.cpp

```diff
@@ -2455,6 +2465,15 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 
 		load_icon_refs(p_preset, main_image, foreground, background);
 
+		if (export_app_bundle == 1 && p_path.find(".aab") == -1) {
+			EditorNode::get_singleton()->show_warning(TTR("Cannot export App Bundle using a *.apk file name."));
+			return ERR_UNCONFIGURED;
+		}
+		if (export_app_bundle == 0 && p_path.find(".aab") != -1) {
+			EditorNode::get_singleton()->show_warning(TTR("Cannot export APK using a *.aab file name."));
+			return ERR_UNCONFIGURED;
+		}
```

Comment:
Similar as above, I'd update the extension being checked to be `.apk` and update the error message along the lines of:
`Invalid filename! Android APK requires the *.apk extension.`